packages/components/src/list/list-item.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import { ENTER } from '@wordpress/keycodes';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Link from '../link';
+
+/**
+ * List component to display a list of items.
+ *
+ * @param {Object} props props for list item
+ */
+function ListItem( props ) {
+	const handleKeyDown = ( event, onClick ) => {
+		if ( typeof onClick === 'function' && event.keyCode === ENTER ) {
+			onClick();
+		}
+	};
+
+	const getItemLinkType = ( item ) => {
+		const { href, linkType } = item;
+
+		if ( linkType ) {
+			return linkType;
+		}
+
+		return href ? 'external' : null;
+	};
+
+	const { item } = props;
```

Comment:
What do you think about deconstructing the object `item`? Maybe that would make this part of the code easier to read.

Reply:
Thanks, done in https://github.com/woocommerce/woocommerce-admin/pull/5892/commits/6e2d0c1310b8ebc17f48b81b22802334608290f2